daphne/http_protocol.py

```diff
@@ -199,16 +199,13 @@ def consume_body(self, buffer_size):
         Consumes body in chunks.
         """
 
-        chunk = self.content.read(buffer_size)
-        if not chunk:
-            self.application_queue.put_nowait({"type": "http.request", "body": chunk})
-            return
-        incomplete_chunk = len(chunk) < buffer_size
-        payload = {"type": "http.request", "body": chunk, "more_body": False}
-        if not incomplete_chunk:
-            payload["more_body"] = True
-        self.application_queue.put_nowait(payload)
-        self.consume_body(buffer_size)
+        while True: 
+            chunk = self.content.read(buffer_size)       
+            more_body = not (len(chunk) < buffer_size)
+            payload = {"type": "http.request", "body": chunk, "more_body": more_body}
+            self.application_queue.put_nowait(payload)
+            if not more_body:
+                break
```

Comment:
So can we get rid of this extra method and just inline at line 189?

Reply:
Done